src/CachePlugin.php

```diff
@@ -67,12 +68,16 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
 
         if ($cacheItem->isHit()) {
             // return cached response
-            return new FulfilledPromise($cacheItem->get());
+            $data = $cacheItem->get();
+            $response = $data['response'];
+            $response = $response->withBody($this->streamFactory->createStream($data['body']));
+
+            return new FulfilledPromise($response);
         }
 
         return $next($request)->then(function (ResponseInterface $response) use ($cacheItem) {
             if ($this->isCacheable($response)) {
-                $cacheItem->set($response)
+                $cacheItem->set(['response' => $response, 'body' => $response->getBody()->__toString()])
```

Comment:
what happens with a stream body when setting the cache? what happens without a stream body? duplicated data in cache? maybe we could check if the body is a stream and if so set it to the cache item, otherwise only set the response? and when restoring check if we have a body in the cache and if we do create the stream from that?